app/src/components/datasets/DatasetContentTabs/Evaluation/AddComparisonModelDialog.tsx

```diff
@@ -0,0 +1,128 @@
+  VStack,
+  Text,
+  Box,
+  Input,
+  Link as ChakraLink,
+} from "@chakra-ui/react";
+import { type ComparisonModel } from "@prisma/client";
+import Link from "next/link";
+
+import { api } from "~/utils/api";
+import { useDataset, useHandledAsyncCallback, useSelectedProject } from "~/utils/hooks";
+import { maybeReportError } from "~/utils/errorHandling/maybeReportError";
+import { getComparisonModelName } from "~/utils/baseModels";
+
+const AddComparisonModelDialog = ({
+  modelId,
+  disclosure,
+  onClose,
+}: {
+  modelId: ComparisonModel | null;
+  disclosure: UseDisclosureReturn;
+  onClose: () => void;
+}) => {
+  const dataset = useDataset().data;
+  const cancelRef = useRef<HTMLButtonElement>(null);
+
+  const selectedProject = useSelectedProject().data;
+  const needsMissingOpenaiKey = !selectedProject?.condensedOpenAIKey;
+
+  const mutation = api.datasets.update.useMutation();
+  const utils = api.useContext();
+
+  const [onUpdateConfirm, updateInProgress] = useHandledAsyncCallback(async () => {
+    if (!dataset?.id || !modelId) return;
+    const resp = await mutation.mutateAsync({
+      id: dataset.id,
+      updates: { enabledComparisonModels: [...dataset.enabledComparisonModels, modelId] },
+    });
+    if (maybeReportError(resp)) return;
+    await utils.datasetEntries.listTestingEntries.invalidate({ datasetId: dataset.id });
+    await utils.datasets.get.invalidate();
+
+    disclosure.onClose();
+  }, [mutation, dataset?.id, dataset?.enabledComparisonModels, modelId, disclosure.onClose]);
+
+  const [modelNameToConfirm, setModelNameToConfirm] = useState("");
+
+  useEffect(() => {
+    if (disclosure.isOpen) {
+      setModelNameToConfirm("");
+    }
+  }, [disclosure.isOpen, setModelNameToConfirm]);
+
+  if (!modelId) return null;
+
+  const modelName = getComparisonModelName(modelId);
+
+  return (
+    <AlertDialog leastDestructiveRef={cancelRef} {...disclosure} onClose={onClose}>
+      <AlertDialogOverlay>
+        <AlertDialogContent>
+          <AlertDialogHeader fontSize="lg" fontWeight="bold">
+            Add Comparison Model
+          </AlertDialogHeader>
+
+          <AlertDialogBody>
+            <VStack spacing={4} alignItems="flex-start">
+              <Text>
+                All existing and future test entries will be evaluated against <b>{modelName}</b>.
+                The OpenAI token associated with this project will be used for these API calls.
+              </Text>
+
+              {needsMissingOpenaiKey ? (
+                <Text>
+                  To include this model, add your OpenAI API key on the{" "}
+                  <ChakraLink as={Link} href="/project/settings" target="_blank" color="blue.600">
+                    <Text as="span">project settings</Text>
+                  </ChakraLink>{" "}
+                  page.
+                </Text>
+              ) : (
+                <VStack>
+                  <Text>To confirm this change, please type the model's ID below.</Text>
```

Comment:
Let's add text like "This will result in 123 calls to the OpenAI API". Right now the dialog does a good job of making this scary but not a great job of giving me the information I need to assess whether I should do it or not.

Reply:
Added better feedback:
<img width="558" alt="Screenshot 2023-10-17 at 12 46 44 PM" src="https://github.com/OpenPipe/OpenPipe/assets/41524992/ac20cd00-5d9b-4105-ba5b-7e24b3c4dc39">